3_harmonize/src/clean_conductivity_data.R

```diff
@@ -0,0 +1,71 @@
+#' @title Clean conductivity data
+#' 
+#' @description 
+#' Function to clean conductivity data, including harmonizing diverse units. 
+#' 
+#' @param wqp_data data frame containing the data downloaded from the WQP, 
+#' where each row represents a data record.
+#' @param char_names_crosswalk data frame containing columns "char_name" and 
+#' "parameter". The column "char_name" contains character strings representing 
+#' known WQP characteristic names associated with each parameter.
+#' @param cond_param_name character string indicating which string in the 
+#' "parameter" column of `char_names_crosswalk` corresponds with conductivity 
+#' data.
+#' 
+#' @returns 
+#' Returns a formatted and harmonized data frame containing data downloaded from 
+#' the Water Quality Portal, where each row represents a data record. Conductivity
+#' units have been standardized to uS/cm where possible. 
+#' 
+clean_conductivity_data <- function(wqp_data, char_names_crosswalk, cond_param_name){
+  
+  # Grab characteristic names associated with conductivity in `char_names_crosswalk`
+  cond_char_names <- char_names_crosswalk %>%
+    filter(parameter == cond_param_name) %>%
+    pull(char_name)
+  
+  # Harmonize conductivity units when possible
+  wqp_data_out <- wqp_data %>%
+    # convert values from mS/cm to uS/cm
+    mutate(ResultMeasureValue = if_else(CharacteristicName %in% cond_char_names & 
+                                          !is.na(ResultMeasureValue) & 
+                                          ResultMeasure.MeasureUnitCode == "mS/cm",
+                                        (ResultMeasureValue * 1000), ResultMeasureValue),
+           # update units in column "ResultMeasure.MeasureUnitCode"
+           ResultMeasure.MeasureUnitCode = case_when(
+             # convert mS/cm to uS/cm
+             CharacteristicName %in% cond_char_names & 
+               !is.na(ResultMeasureValue) & 
```

Comment:
I think it would clean things up if you took these two logicals (`CharacteristicName %in% cond_char_names & !is.na(ResultMeasureValue)` and collapsed into a new column since they are used over and over. Then drop the column afterwards. Something like

```r
wqp_data_out <- wqp_data %>%
    mutate(is_nonNA_conductivity = CharacteristicName %in% cond_char_names & 
                                          !is.na(ResultMeasureValue)) %>%
    # convert values from mS/cm to uS/cm
    mutate(ResultMeasureValue = if_else(is_nonNA_conductivity & 
                                          ResultMeasure.MeasureUnitCode == "mS/cm",
                                        (ResultMeasureValue * 1000), ResultMeasureValue),
           # update units in column "ResultMeasure.MeasureUnitCode"
           ResultMeasure.MeasureUnitCode = case_when(
             # convert mS/cm to uS/cm
             is_nonNA_conductivity &
                 ResultMeasure.MeasureUnitCode == "mS/cm" ~ "uS/cm",
             # convert equivalent units (umho/cm) to uS/cm
             is_nonNA_conductivity & 
               ResultMeasure.MeasureUnitCode == "umho/cm" ~ "uS/cm"
```

Reply:
That could be good in the temperature fxn, too

Reply:
Great idea, I like how your suggested change reduces the amount of duplicated code within these functions. Edits made in https://github.com/USGS-R/ds-pipelines-targets-example-wqp/pull/75/commits/7cd319f3617007f62e0eb5654eddf4f4944647da and https://github.com/USGS-R/ds-pipelines-targets-example-wqp/pull/75/commits/917a4177760881865047804d3a55d510b0b6c7ca.